acasclient/acasclient.py

```diff
@@ -2324,9 +2324,15 @@ def _get_meta_lot_by_parent_corp_name(self, parent_corp_name):
         """
         # Look up the compound to find a lot, so we can access a MetaLot
         search_results = self.cmpd_search(corpNameList=parent_corp_name)
-        if len(search_results['foundCompounds']) == 0:
+
+        found_compounds = search_results['foundCompounds']
+
+        # Return the first found_compound which matches the parent_corp_name
+        found_parent = next((x for x in found_compounds if x['corpName'] == parent_corp_name), None)
+
+        if not found_parent:
             raise RuntimeError(f'Parent corp name {parent_corp_name} could not be found')
-        lot_corp_name = search_results['foundCompounds'][0]['lotIDs'][0]['corpName']
```

Comment:
The bug was here, we assumed the first in the list was an exact match but never verified it.